ProcessLib/ProcessVariable.h

```diff
@@ -56,10 +57,20 @@ class ProcessVariable
 	/// Returns a mesh on which the process variable is defined.
 	MeshLib::Mesh const& getMesh() const;
 
-	void initializeDirichletBCs(MeshGeoToolsLib::MeshNodeSearcher& searcher,
-	        const AssemblerLib::LocalToGlobalIndexMap& dof_table,
-	        const unsigned nodal_dof_idx,
-	        std::vector<GlobalIndexType>& global_ids, std::vector<double>& values);
+	template <typename OutputIterator>
+	void initializeDirichletBCs(
+	    OutputIterator output_bcs,
+	    MeshGeoToolsLib::MeshNodeSearcher& searcher,
+	    const AssemblerLib::LocalToGlobalIndexMap& dof_table,
+	    const unsigned nodal_dof_idx)
+	{
+		for (auto& bc_config : _dirichlet_bc_configs)
+		{
+			DirichletBc<GlobalIndexType> bc;
+			bc_config->initialize(searcher, dof_table, nodal_dof_idx, bc);
+			output_bcs = bc;  // output iterator
```

Comment:
Thanks Christoph. There is a mistake here. But it works!
How? I think because operator= call on std::back_inserter makes a push_back on the underlying container, this works. But it would not (as pointed out in the SO answer above) for pointers, which are also OutputIterator's.

Will fix it, rebase and merge.
@chleh OK?